### PR TITLE
Implement an endpoint to update a todo item in the DB by its ID

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**DESCRIPTION OF CHANGES**

- Refactor create-todos-dto to todos.dto, and export UpdateTodosDto from here
- Extract the id from the URL param and body (based on the data shape defined by the UpdateTodosDto) from the request body and send it to the service
- Fetch the todo item based on the id using existing getTodoById endpoint that has validation as well
- Assign any changes if any to the todo item
- Flush

Apologies for the video quality, had to downgrade it as limit is only 10mb

https://github.com/user-attachments/assets/92b89a6c-4459-4038-af65-6beba7beebcb

### CODE REVIEW

File: src/todos/todos.service.ts [hunk 1]
```diff
@@ -22,4 +22,11 @@ export class TodosService {
     }
     return foundTodo;
   }
+
+  async updateTodoById(id: string, updateTodo: UpdateTodoDto): Promise<Todo> {
+    const foundTodo = await this.getTodoById(id);
+    this.todoRepository.assign(foundTodo, updateTodo);
+    await this.todoRepository.getEntityManager().flush();
```
Comment (ishmam-iqbal-sazim) [inline src/todos/todos.service.ts]: In my opinion, this is a case where a repository method (like updateOne) makes sense as the builtin method names are not exactly clear

File: src/todos/todos.service.ts [hunk 2]
```diff
@@ -22,4 +22,11 @@ export class TodosService {
     }
     return foundTodo;
   }
+
+  async updateTodoById(id: string, updateTodo: UpdateTodoDto): Promise<Todo> {
```
Comment (ishmam-iqbal-sazim) [inline src/todos/todos.service.ts]:
```suggestion
  async updateOne(id: string, updateTodo: UpdateTodoDto): Promise<Todo> {
```

No need to specify `todo` as this is in todosService.

File: src/todos/todos.dto.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  @IsNotEmpty()
+  @IsString()
+  description!: string;
+
+  @IsNotEmpty({ message: 'Due date is required.' })
+  @IsDateString({ strict: true })
+  @Matches(/^\d{4}-\d{2}-\d{2}$/, {
+    message: 'Invalid date format. Use YYYY-MM-DD.',
+  })
```
Comment (ishmam-iqbal-sazim) [inline src/todos/todos.dto.ts]:
```suggestion
  @IsNotEmpty({ message: 'Due date is required.' })
  @IsDate()
  @Type(() => Date)
```

Can't we do something like this? I am not sure if we need this level of accuracy for dates. Same comment for other instances like this throughout this PR

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

**Comment by yamin-faruque-sazim** [comment]
Addressed all comments

**Pull request merged**